packages/core/src/state/snapshots.ts

```diff
@@ -0,0 +1,14 @@
+import { useSnapshot } from "valtio";
+import { Web3InboxClient } from "../client";
+
+export const useSubscriptionState = () => {
```

Comment:
Not sure if those `useSnapshot`s belong here. This is generally used in React, when a state variable is used in the render function.

Reply:
I wish we could but at the moment it would be too difficult to circumvent this:  https://github.com/pmndrs/valtio/discussions/706